### ISSUE TITLE
Unit testing in Angular with ng-lottie element

### ISSUE THREAD

**Issue by kzndzhiro** [ngx-lottie/ngx-lottie#36]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[x] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

## What is this request about?
@arturovt Thank you for developing this library. It is really great and easy to work with. Excuse me if my more of a support request does not belong here and should be deleted.

One issue I am facing in my current setup is when I test Lottie in a *ngIf directive. In this case, when the animation should be displayed but really quickly I change the binding of the *ngIf to be false I have the LottieEventFacade calling dispose method which calls destroy on an animatiionItem of null
`private dispose(): void {
    if (isPlatformServer(this.platformId)) {
      return;
    }

    // destroy() will remove all events listeners
    this.animationItem!.destroy(); // in here this.animationItem is not shown.
    this.animationItem = null;
  }`
And later in time the loadAnimation method is triggered.

Since I am not proficient with testing and what to mock in this case, I currently have added CUSTOM_ELEMENTS_SCHEMA to the tests so I can go around the issue.
Another way around for me was to Provide the PLATFORM_ID token as "server" because then the destroy is not called.

All in all my question is how is the best approach to test this scenario:
1. Lottie directive starts
2. The ng-lottie element is destroyed before it has loaded and initialized the animationItem
3. An error is thrown that cannot call destroy() of null

**Comment by arturovt** [comment]
@kzndzhiro 

Thank you for the issue!

You scenario actually gave me a thought that it could have been a bug, but I'm not sure. Could you post some code here? I will try to reproduce this behaviour locally.

**Comment by kzndzhiro** [comment]
@arturovt  
Thank you for your quick response and interest.
I will try to set up a demo link tomorrow and post it here.

**Comment by kzndzhiro** [comment]
HI @arturovt 
Sorry for the delay. I have not forgotten about the link to reproduce the issue. I failed to reproduce the issue in a simple angular app so far. I will check our app setup and get back to you soon.
If I see that we have an issue with our app setup I will close the issue.

**Comment by arturovt** [comment]
Hey. I was able to reproduce your issue like that:
```ts
@ViewChild('container', { static: true, read: ViewContainerRef }) container!: ViewContainerRef;

constructor(
  private ngZone: NgZone,
  private resolver: ComponentFactoryResolver
) {}

ngOnInit() {
  const factory = this.resolver.resolveComponentFactory(LottieComponent);
  const ref = this.container.createComponent(factory);
  ref.destroy();
}
```

But if I add `ref.changeDetectorRef.detectChanges()` before `ref.destroy()` then it works like a charm because it runs the change detection on this component and invokes `ngOnInit` (where `animationItem` is set).

I will need to add a check for `!== null`.

**Comment by arturovt** [comment]
I have published `5.3.1`. Could you check it up?

**Comment by kzndzhiro** [comment]
@arturovt  
You are awesome 🥇 
Thank you for fixing it up!
It does work like a charm now.
